### PR TITLE
[aes] Simplify clearing, clear input data regs with pseudo-random data instead of zero

### LINKED ISSUE #4653: [aes] Simplify clearing functionality
Currently, aes features several bits in the trigger register to separately clear
- key registers,
- iv registers,
- data input registers, and
- data output registers.

@rasmus-madsen just told me that from a verification standpoint this is bad. There are many different corner cases to test (e.g. clear key without iv). Also, some of them are very difficult to test. For example the iv registers are updated internally by the module. Thus, clearing the key after having computed a first block and then re-providing the same key and data will not give the same output. Also, from a software viewpoint, there is probably very little value in allowing the module to clear the key registers but not the iv registers.

For this reason, @rasmus-madsen and I suggest to simplify the clearing functionality. We see two possible options:
1. A trigger single bit to clear everything in one shot: key, iv, data in, data out registers.
2. Two trigger bits to separately clear:
    i) key, iv, and data in registers
    ii) data out registers

Option 2. has the advantage that software - after starting the last block of a message - can already trigger the first part of the clearing i). This will start immediately after finishing the last block but not touch the data out registers. Software has time to read out the last data block and then clears them after that ii).

Before implementing this change, I wanted to quickly get your opinion @cdgori , @tjaychen , @moidx . Do you see a problem with that?

### PR DESCRIPTION
**Pull request by vogelpi**
This PR contains two main changes:
1. Simplify the clearing functionality by reducing the number of trigger bits from 4 to 2:
- One bit for clearing key, IV, and input data registers, and
- one bit for clearing output data registers.

This way, software can already trigger the clearing of the key, IV, and input data registers while the last block operation is still running. The After the operation is finished, the output registers can be read in parallel to the clearing operation. Having more, separate clearing triggers is not useful but makes design verification a lot harder as it adds many difficult corner cases.

This resolves #4653.

2. Clear input data regs with pseudo-random data (as stated in the documentation) instead of zero.

### CODE REVIEW

**Comment by vogelpi** [comment]
FYI @silvestrst

File: hw/ip/aes/data/aes.hjson [hunk 1]
```diff
@@ -429,46 +429,30 @@
...
+        resval: "0"
         desc:  '''
           Keep AES unit paused (0) or trigger the encryption/decryption of one data block (1).
           This trigger is ignored if MANUAL_OPERATION=0 (see Control Register).
         '''
         tags: ["excl:CsrNonInitTests:CsrExclWriteCheck"]
       }
       { bits: "1",
-        name: "KEY_CLEAR",
+        name: "KEY_IV_DATA_IN_CLEAR",
```
Comment (rasmus-madsen) [inline hw/ip/aes/data/aes.hjson:440]: Im ok with this name, but it could be something simpler like
Input_regs_clear

File: hw/ip/aes/dv/env/seq_lib/aes_base_vseq.sv [hunk 2]
```diff
@@ -71,9 +71,9 @@ class aes_base_vseq extends cip_base_vseq #(
     `uvm_info(`gfn, $sformatf("%s",txt), UVM_MEDIUM)
 
     ral.trigger.set(0);
-    ral.trigger.key_clear.set(clr_vector.key);
-    ral.trigger.iv_clear.set(clr_vector.iv);
-    ral.trigger.data_in_clear.set(clr_vector.data_in);
+    ral.trigger.key_iv_data_in_clear.set(clr_vector.key);
+    ral.trigger.key_iv_data_in_clear.set(clr_vector.iv);
+    ral.trigger.key_iv_data_in_clear.set(clr_vector.data_in);
```
Comment (rasmus-madsen) [inline hw/ip/aes/dv/env/seq_lib/aes_base_vseq.sv:76]: These look like search an replace output
please update to 
```suggestion
    ral.trigger.key_iv_data_in_clear.set(clr_vector.data_in);
```

I will update the clr_vector to be a 2 bit with real names once this PR goes in

**Review by rasmus-madsen — CHANGES_REQUESTED** [review]
Looks good with one change request

File: hw/ip/aes/data/aes.hjson (hunk 1 above)
Comment (vogelpi) [reply hw/ip/aes/data/aes.hjson:440]: I was thinking about that, too. But my fear is that people might erroneously think `INPUT_REGS_CLEAR` only deals with input data. Thus I thought it would be better to be as explicit as possible.

Comment (rasmus-madsen) [reply hw/ip/aes/data/aes.hjson:440]: that it fine with me

File: hw/ip/aes/dv/env/seq_lib/aes_base_vseq.sv (hunk 2 above)
Comment (vogelpi) [reply hw/ip/aes/dv/env/seq_lib/aes_base_vseq.sv]: Done!

**Comment by vogelpi** [comment]
Thanks @rasmus-madsen for your review, I have rebased on master and made the change you requested inside `aes_base_vseq.sv `.

**Review by rasmus-madsen — APPROVED** [review]
thank you for the updates
LGTM

**Review by cdgori — COMMENTED** [review]
I realize this is post-merge but the concept of merging the input data clears is a good one, and I don't see any issues.
